File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/menu/util/MenuItemConfig_Visual.cpp
#include "MenuItemConfig_Visual.h"

MenuItemConfig_Visual::MenuItemConfig_Visual(
    std::string itemID,
    UiId* parentComposedID,
    VisualCreateConfig* visualCreateConfig, VisualPropertiesConfig* itemProperties)
    : super(itemID),
      parentComposedID(parentComposedID),
      visualCreateConfig(visualCreateConfig), itemProperties(itemProperties)
{
	if (itemID.size() == 0) {
		throw LogicException(LOC);
	}
}

MenuItemConfig_Visual::MenuItemConfig_Visual(
    std::string itemID,
    std::string parentComposedID,
    VisualCreateConfig* visualCreateConfig, VisualPropertiesConfig* itemProperties)
    : MenuItemConfig_Visual(
        itemID,
        UiId::newIdFromComposedId(parentComposedID, "_local"),
        visualCreateConfig, itemProperties)
{
    //void
}

VisualPropertiesConfig* MenuItemConfig_Visual::getItemProperties() {
    return itemProperties;
}

MenuItemConfig_Visual::~MenuItemConfig_Visual                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           () {
    //void
}

/*
asdAA;


public MenuItemConfig_Visual(String itemID, VisualCreateConfig visualCreateConfig, VisualPropertiesConfig itemProperties) {
    //this(itemID, null, visualCreateConfig, itemProperties);

    super(itemID);

    this.parentVisualID = null;

    this.visualCreateConfig = visualCreateConfig;

    this.itemProperties = itemProperties;
}
*/

File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/base/menu/util/MenuItemConfig_Visual.java
package base.menu.util;

import base.visual.util.VisualCreateConfig;
import base.visual.util.VisualPropertiesConfig;

public class MenuItemConfig_Visual extends MenuItemConfig implements IMenuItemConfig_DynamicProperties {

    public String[] parentVisualID;

    public VisualCreateConfig visualCreateConfig;

    private VisualPropertiesConfig itemProperties;

	public MenuItemConfig_Visual(String itemID, String[] parentVisualID, VisualCreateConfig visualCreateConfig, VisualPropertiesConfig itemProperties) {
		super(itemID);

        this.parentVisualID = parentVisualID;

        this.visualCreateConfig = visualCreateConfig;

        this.itemProperties = itemProperties;
	}

    @Override
    public VisualPropertiesConfig getItemProperties() {
        return itemProperties;
    }

}
